Credit Card Validation/src/main/java/CreditCardValidator.java

```diff
@@ -1,94 +1,65 @@
-import java.util.ArrayList;
-import java.util.List;
-
-enum CardType {
-  VISA("VISA", "4"),
-  JCB("JCB", "35"),
-  DISCOVER("DISCOVER", "6011", "65"),
-  MASTERCARD("MASTERCARD", "51", "52", "53", "54", "55"),
-  AMERICAN_EXPRESS("AMERICAN EXPRESS", "34", "37");
-  
-  private final String paymentName;
-  private final List<String> prefix;
-  
-  CardType(String paymentName, String ...prefix) {
-    this.paymentName = paymentName;
-    this.prefix = new ArrayList<String>(List.of(prefix));
-  }
-  
-  public List<String> getPrefix() {
-    return this.prefix;
-  }
-  public String getPaymentName() {
-    return this.paymentName;
-  }
-}
-
 public class CreditCardValidator {
   private final int CREDIT_CARD_LENGTH = 16;
-  private final String creditCardNumber;
 
-  public CreditCardValidator(String creditCardNumber) {
-    this.creditCardNumber = creditCardNumber;
+  public String validateNumbers(String creditCardNumber) {
+    boolean letterFound = hasLetter(creditCardNumber);
+
+    if (!letterFound) {
+      String[] cardDigitsArray = creditCardNumber.split(" ");
+      if (cardDigitsArray.length == 4) {
+        return PaymentSystemResolver(cardDigitsArray[0]);
+      } else if (creditCardNumber.length() == 16) {
+        return PaymentSystemResolver(creditCardNumber.substring(0, 4));
+      } else {
+        return "";
+      }
+    }
+    return "";
   }
 
-  public String validateNumbers() {
+  private boolean hasLetter(String creditCardNumber) {
     for (int i = 0; i < creditCardNumber.length(); i++) {
       if (Character.isWhitespace(creditCardNumber.charAt(i)))
         continue;
 
       if (!Character.isDigit(creditCardNumber.charAt(i))) {
```

Comment:
It means that this method checks not only letters - symbols are also will return true. Wrong name of method.